engine/src/main/java/org/destinationsol/game/asteroid/AsteroidBuilder.java

```diff
@@ -32,20 +33,47 @@
 import org.destinationsol.game.drawables.Drawable;
 import org.destinationsol.game.drawables.DrawableLevel;
 import org.destinationsol.game.drawables.RectSprite;
+import org.json.JSONObject;
+import org.terasology.assets.ResourceUrn;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Set;
 
 public class AsteroidBuilder {
     private static final float DENSITY = 10f;
     private static final float MAX_A_ROT_SPD = .5f;
     private static final float MAX_BALL_SZ = .2f;
     private final CollisionMeshLoader collisionMeshLoader;
     private final List<TextureAtlas.AtlasRegion> textures;
+    private final String moduleName;
 
-    public AsteroidBuilder() {
-        collisionMeshLoader = new CollisionMeshLoader("engine:asteroids");
-        textures = Assets.listTexturesMatching("engine:asteroid_.*");
+    public AsteroidBuilder(String shipName) {
+        moduleName = getModule(shipName);
+        CollisionMeshLoader tempCML;
+        List<TextureAtlas.AtlasRegion> tempTextures;
+        try {
+            tempCML = new CollisionMeshLoader(moduleName +":asteroids");
+            tempTextures = Assets.listTexturesMatching(moduleName +":asteroid_.*");
+        } catch (RuntimeException e) {
+            tempCML = new CollisionMeshLoader("engine:asteroids");
+            tempTextures = Assets.listTexturesMatching("engine:asteroid_.*");
+        }
+        collisionMeshLoader = tempCML;
+        textures = tempTextures;
+    }
+
+    private String getModule(String shipName) {
+        Set<ResourceUrn> configUrnList = Assets.getAssetHelper().list(Json.class, "[a-zA-Z]*:playerSpawnConfig");
+
+        for (ResourceUrn configUrn : configUrnList) {Json json = Assets.getJson(configUrn.toString());
```

Comment:
New line for the statement

Reply:
Sorry, I'm a bit confused by what you mean here?

Reply:
Add a new line after the curly braces of the for loop instead of directly starting with the statement.